### DISCUSSION TITLE
Certification error when starting hosting repo with caddy

### DISCUSSION THREAD

**Discussion by srameshr** [plausible/analytics#661] [Self-Hosted Support] [answered]
```
docker-compose -f docker-compose.yml -f reverse-proxy/docker-compose.caddy-gen.yml up
```

Running with reverse proxy a few times always fails with cert error. Can we somehow use a local certificate? 
```
2021/01/28 12:05:31 [ERROR] attempt 1: [xyz.xom] Obtain: [xyz.com] acme: error: 429 :: POST :: https://acme-v02.api.letsencrypt.org/acme/new-order :: urn:ietf:params:acme:error:rateLimited :: Error creating new order :: too many certificates already issued for exact set of domains: xyz.com: see https://letsencrypt.org/docs/rate-limits/, url:  - retrying in 1m0s
```

**Comment by Vigasaurus** [comment]
Certificate serving is purely on the side of your reverse proxy; you can try adding the cert through certbot without the docker containers running, and instead with just a basic site, or you can generate just a cert and add it to the reverse proxy config manually if that's not working for you. 

The error you're getting is between Caddy and LetsEncrypt - and it's that you've somehow requested too many certificates for that domain recently. I'd recommend LetsEncrypt support. 

https://letsencrypt.org/docs/rate-limits/

**Reply by srameshr** [reply]
Yes, I get that. But, I was wondering if we could make changes to the caddy docker file that picks up the certificate from the local drive instead of requesting it from letsencrypt every time it's run.

**Comment by srameshr** [comment, marked as answer]
@Vigasaurus So, I figured out the issue. I was running docker-compose down instead of stop. This was removing the volumes associated with caddy along with the certificates inside the volume.
